File: C++/dpr.cpp
#include <iostream>
#include <string>

using namespace std;

class Human
{
	private: // atribut untuk class human
		string name;
		string id;
		string nama_bidang;
		string nama_partai;
	
	public: 
	
		Human()
		{
			this->name = "";
		}
		
		list<Human> llist;
		
		Human(string name, string id, string nama_bidang, string nama_partai)
		{
			this->name = name;
			this->id = id;
			this->nama_bidang = nama_bidang;
			this->nama_partai = nama_partai;
		}
		
		string get_name() // mengambil nama
		{
			return this->name;
		}
		
		
		void set_name(string name) // membuat/set nama
		{
			this->name = name;
		}
		
		void set_id(string id) // membuat/set id
		{
			this->id= id;
		}
		
		string get_id() // mengambil id
		{
			return this->id;
		}
		
		string get_namaBidang() // mengambil nama bidang
		{
			return this->nama_bidang;
		}
		
		void set_namaBidang(string nama_bidang) // membuat/set nama bidang
		{
			this->nama_bidang = nama_bidang;
		}
		
		string get_namaPartai() // mengambil nama partai
		{
			return this->nama_partai;
		}
		
		void set_namaPartai(string nama_partai) // membuat/set nama partai
		{
			this->nama_partai = nama_partai;
		}
		
		void renameHuman(string id,string namaBaru) // edit nama anggota
		{
			for(list<Human>:: iterator it = llist.begin(); it != llist.end(); ++it )
			{
				if(it->get_id() == id) // jika ditemukan sesuai dengan id
				{
					cout << "Nama " << it->get_name() << " Telah diubah menjadi " << namaBaru << endl;
					it->set_name(namaBaru);
				}
			}
		}
		
		void renameBidang(string id,string bidangBaru) // edit nama bidang
		{
			for(list<Human>:: iterator it = llist.begin(); it != llist.end(); ++it )
			{
				if(it->get_id() == id) // jika ditemukan sesuai dengan id
				{
					cout << "Nama " << it->get_namaBidang() << " Telah diubah menjadi " << bidangBaru << endl;
					it->set_namaBidang(bidangBaru);
				}
			}
		}
		
		void renamePartai(string id,string partaiBaru) // edit nama partai
		{
			for(list<Human>:: iterator it = llist.begin(); it != llist.end(); ++it )
			{
				if(it->get_id() == id) // jika ditemukan sesuai dengan id
				{
					cout << "Nama " << it->get_namaPartai() << " Telah diubah menjadi " << partaiBaru << endl;
					it->set_namaPartai(partaiBaru);
				}
			}
		}
		
		void tambah() // input anggota
		{
			int n; // deklarasi
			cout << "Masukkan berapa orang?: ";
			cin >> n; // input
			for(int i = 0; i < n; i++)
			{
				Human temp;
				cout << "No " << (i + 1) << endl; 
				cout << "Masukkan nama: "; // membuat nama
				cin >> name;
				temp.set_name(name);
				
				cout << "Masukkan ID: "; // membuat id
				cin >> id;
				temp.set_id(id);
				
				cout << "Masukkan nama Bidang: "; // membuat nama bidang
				cin >> nama_bidang;
				temp.set_namaBidang(nama_bidang);
				
				cout << "Masukkan nama Partai: "; // membuat nama partai
				cin >> nama_partai;
				temp.set_namaPartai(nama_partai);
				
				llist.push_back(temp);
			}
		}
		
		
		int ketemu(string id) // untuk menemukan id anggota
		{
			for(list<Human>:: iterator it = llist.begin(); it != llist.end(); ++it)
			{
				if(it->get_id() == id)
				{
					return 1; // jika ada
				}
			}
			return 0; // jika tidak ada
		}
		
		void hapus(string id) // menghapus anggota
		{
			for(list<Human>:: iterator it = llist.begin(); it != llist.end(); )
			{
				if(it->get_id() == id) // jika ditemukan
				{
					cout << it->get_name() << " Telah dihapus!" << endl;
					it = llist.erase(it); // hapus dengan function 'erase'
				}
				else
				{
					++it; // iterasi jika tidak ditemukan
				}
			}
		}
		
		void tampil() // menampilkan anggota
		{
			// print all element
			cout << "Anggota: " << '\n';
			int i = 0;
			for(list<Human>::iterator it = llist.begin();it != llist.end() ;it++)
			{
				cout << (i + 1) << ". " << it->get_name() << " || " << it->get_id() << " || " << it->get_namaBidang() << " || " << it->get_namaPartai() << '\n';
				i++;
			}
			cout << endl;
		}
		
		~Human() // destruct
		{
			
		}
};

File: Php/program/Dpr.php
<?php
class Dpr
{
	private $nama = '';
	private $id = '';
	private $nama_bidang = '';
	private $nama_partai = '';
	private $foto;
	
	# construct
	public function __construct($name = '', $id = '', $nama_bidang = '', $nama_partai = '')
	{
		$this->nama = $name;
		$this->id= $id;
		$this->nama_bidang = $nama_bidang;
		$this->nama_partai = $nama_partai;
	}
	
	# setter
	public function setName($name)
	{
		$this->nama = $name;
	}
	
	public function setId($id)
	{
		$this->id = $id;
	}
	
	public function setBidang($nama_bidang)
	{
		$this->nama_bidang = $nama_bidang;
	}
	
	public function setPartai($nama_partai)
	{
		$this->nama_partai = $nama_partai;
	}
	public function setFoto($foto)
	{
		$this->foto = $foto;
	}
	
	# getter
	public function getName()
	{
		return $this->nama;
	}
	
	public function getId()
	{
		return $this->id;
	}
	
	public function getBidang()
	{
		return $this->nama_bidang;
	}
	
	public function getPartai()
	{
		return $this->nama_partai;
	}
	
	public function getFoto()
	{
		return $this->foto;
	}
	
	public function Edit(&$list, $id, $name, $banding)
	{
		$bidang = $name; $partai = $name; $foto = $name; // memasukkan setiap nama ke variable
		$NamaAda = 0; $PartaiAda = 0; $BidangAda = 0; $FotoAda = 0; // marking
		foreach($list as $humans)
		{
			if($humans->getId() == $id) // jika id tersebut memang terdapat dalam daftar
			{
				if(strcmp($banding, "Nama") == 0) // jika user pengen mengubah nama
				{
					$NamaAda = 1; // jika terdapat dalam daftar, maka
					if($NamaAda == 1) // akan menampilkan jika nama diubah
					{
						echo "Nama " . $humans->getName() . " Menjadi " . $name;
					}
					$humans->setName($name); // masuk ke function setter
				}
				else if(strcmp($banding, "Bidang") == 0) // jika user pengen mengubah bidang
				{
					$BidangAda = 1; // jika terdapat dalam daftar, maka
					if($BidangAda == 1) // akan menampilkan jika bidang diubah
					{
						echo "Bidang " . $humans->getBidang() . " Menjadi " . $bidang;
					}
					$humans->setBidang($bidang); // masuk ke function setter
				}
				else if(strcmp($banding, "Partai") == 0) // jika user pengen mengubah partai
				{
					$PartaiAda = 1; // jika terdapat dalam daftar, maka
					if($PartaiAda == 1) // akan menampilkan jika partai diubah
					{
						echo "Partai " . $humans->getPartai() . " Menjadi " . $partai;
					}
					$humans->setPartai($partai);
				}
				else if(strcmp($banding, "Foto") == 0) // jika user pengen mengubah foto
				{
					$FotoAda = 1; // jika terdapat dalam daftar, maka
					if($FotoAda == 1) // akan memberitahu jika foto diubah berhasil
					{
						echo "Foto dirubah!";
					}
					$humans->setFoto($foto); // masuk ke function setter
				}
			}
		}
		
		echo '</br>';
	}
	
	public function ketemu($list, $id) // mencari id anggota yang dicari
	{
		foreach($list as $humans)
		{
			if($humans->getId() == $id)
			{
				return 1;
			}
		}
		return 0;
	}
	
	public function hapus(&$list, $id) // fungsi menghapus anggota
	{
		foreach($list as $cari => $humans)
		{
			if($humans->getId() == $id)
			{
				echo '<i><b>'. $humans->getName(). " telah dihapus!" .'</i></b>';
				unset($list[$cari]);
			}
		}
		echo '</br>';
	}
	
	public function tampil($list)  // menampilkan anggota
	{
		if($list != null && !empty($list)) // jika list tidak kosong
		{
			echo "<table border='1'>";
			echo "<tr><th>Nama</th><th>Id</th><th>Bidang</th><th>Partai</th><th>Foto</th></tr>";
			foreach ($list as $human)
			{
				echo "<tr>";
				echo "<td>" . $human->getName() . "</td>";
				echo "<td>" . $human->getId() . "</td>";
				echo "<td>" . $human->getBidang() . "</td>";
				echo "<td>" . $human->getPartai() . "</td>";
				echo "<td><img src='" . $human->getFoto() . "' width='30' height='30' alt='Foto profil'></td>";
				echo "</tr>";
			}
			echo "</table>";
		}
		else // jika list kosong
		{
			echo "Kosong!\n";
		}
	}
}
?>